app/src/main/dist/bin/crate

```diff
@@ -34,81 +32,139 @@
-# /usr/local/share/crate/crate.in.sh, or
-# /opt/crate/crate.in.sh and make your modifications there.
-#
-# Another option is to specify the full path to the include file in the
-# environment. For example:
-#
-#   $ CRATE_INCLUDE=/path/to/in.sh crate -p /var/run/crate.pid
-#
-# Note: This is particularly handy for running multiple instances on a
-# single installation, or for quick tests.
-#
-# If you would rather configure startup entirely from the environment, you
-# can disable the include by exporting an empty CRATE_INCLUDE, or by
-# ensuring that no include files exist in the aforementioned search list.
-# Be aware that you will be entirely responsible for populating the needed
-# environment variables.
 
 CDPATH=""
 SCRIPT="$0"
 
 # SCRIPT may be an arbitrarily deep series of symlinks. Loop until we have the concrete path.
 while [ -h "$SCRIPT" ] ; do
-  ls=`ls -ld "$SCRIPT"`
+  ls=$(ls -ld "$SCRIPT")
   # Drop everything prior to ->
-  link=`expr "$ls" : '.*-> \(.*\)$'`
+  link=$(expr "$ls" : '.*-> \(.*\)$')
   if expr "$link" : '/.*' > /dev/null; then
     SCRIPT="$link"
   else
-    SCRIPT=`dirname "$SCRIPT"`/"$link"
+    SCRIPT=$(dirname "$SCRIPT")/"$link"
   fi
 done
 
 # determine crate home
-CRATE_HOME=`dirname "$SCRIPT"`/..
+CRATE_HOME=$(dirname "$SCRIPT")/..
 
 # make CRATE_HOME absolute
-CRATE_HOME=`cd "$CRATE_HOME"; pwd`
-
-# If an include wasn't specified in the environment, then search for one...
-if [ "x$CRATE_INCLUDE" = "x" ]; then
-    # Locations (in order) to use when searching for an include file.
-    for include in /usr/share/crate/crate.in.sh \
-                   /usr/local/share/crate/crate.in.sh \
-                   /opt/crate/crate.in.sh \
-                   ~/.crate.in.sh \
-                   `dirname "$0"`/crate.in.sh; do
-        if [ -r "$include" ]; then
-            . "$include"
-            break
-        fi
-    done
-# ...otherwise, source the specified include.
-elif [ -r "$CRATE_INCLUDE" ]; then
-    . "$CRATE_INCLUDE"
+CRATE_HOME=$(cd "$CRATE_HOME"; pwd)
+
+
+if [ "x$CRATE_CLASSPATH" != "x" ]; then
+    cat >&2 << EOF
+Error: Don't modify the classpath with CRATE_CLASSPATH.
+Add plugins and their dependencies into the plugins/ folder instead.
+EOF
+    exit 1
+fi
+
+for libname in "$CRATE_HOME"/lib/*.jar; do
+    if [ "x$CRATE_CLASSPATH" != "x" ]; then
+        CRATE_CLASSPATH="$CRATE_CLASSPATH:$libname"
+    else
+        CRATE_CLASSPATH="$libname"
+    fi
+done
+
+if [ "$CRATE_MIN_MEM" = "" ]; then
+    CRATE_MIN_MEM=256m
+fi
+if [ "x$CRATE_HEAP_SIZE" != "x" ]; then
+    CRATE_MIN_MEM=$CRATE_HEAP_SIZE
+    CRATE_MAX_MEM=$CRATE_HEAP_SIZE
+fi
+
+# min and max heap sizes should be set to the same value to avoid
+# stop-the-world GC pauses during resize, and so that we can lock the
+# heap in memory on startup to prevent any of it from being swapped
+# out.
+JAVA_OPTS="$JAVA_OPTS -Xms${CRATE_MIN_MEM}"
+if [ "x$CRATE_MAX_MEM" != "x" ]; then
+    JAVA_OPTS="$JAVA_OPTS -Xmx${CRATE_MAX_MEM}"
+fi
```

Comment:
I wonder if we should clean this up further to get rid of CRATE_MIN_MEM and CRATE_MAX_MEM. This logic supports that only CRATE_MIN_MEM is set but this would fail later on the `BootstrapChecks` which requires min/max values set to the same value. Or at least set `-Xmx` to the same value of CRATE_MIN_MEM if neither CRATE_HEAP_SIZE nor CRATE_MAX_MEM is set?